### PR TITLE
opa 0.8.2 (new formula)

### PR DESCRIPTION
**Pull request by billyogat**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by billyogat** [comment]
Looks like a new formula isn't supposed to have a `--HEAD` option: https://github.com/Homebrew/brew/pull/4071. 

I'm not sure why, and I wasn't able to find anything in the published documentation (https://docs.brew.sh/Formula-Cookbook#unstable-versions-devel-head).

File: Formula/opa.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  desc "Open source, general-purpose policy engine"
+  homepage "https://www.openpolicyagent.org"
+  url "https://github.com/open-policy-agent/opa/archive/v0.8.2.tar.gz"
+  sha256 "475cecc8cc12bddfe6d58311e3e084fb1583647b1902c8955d45adba1d22a273"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["PATH"] = ENV["PATH"] + ":" + buildpath + "/bin"
```
Comment (DomT4) [inline Formula/opa.rb]: `ENV.append_path "PATH", buildpath/"bin"`

File: Formula/opa.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    dir = buildpath/"src/github.com/open-policy-agent/opa"
+    dir.install Dir["*"]
+
+    system "make", "deps", "-C", dir
```
Comment (DomT4) [inline Formula/opa.rb]: Probably easier to just `cd` into `dir` rather than calling `dir` 5 times in 5 lines.

File: Formula/opa.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    system "make", "-C", dir
+
+    opa_bin = dir/"opa_darwin_amd64"
+    mv opa_bin, dir/"opa"
+    bin.install(dir/"opa")
+  end
+
+  test do
+    system "echo", "pi = 3.14", "|", "#{bin}/opa", "eval", "--stdin"
```
Comment (DomT4) [inline Formula/opa.rb]: Use `pipe_output()` here.

File: Formula/opa.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    ENV["GOPATH"] = buildpath
+    ENV["PATH"] = ENV["PATH"] + ":" + buildpath + "/bin"
+
+    dir = buildpath/"src/github.com/open-policy-agent/opa"
+    dir.install Dir["*"]
+
+    system "make", "deps", "-C", dir
+    system "make", "-C", dir
+
+    opa_bin = dir/"opa_darwin_amd64"
```
Comment (DomT4) [inline Formula/opa.rb]: `bin.install "opa_darwin_amd64" => "opa"`, although arguably here either we should be building it ourselves without the `Makefile` or upstream should be providing a `make install` step.

**Pull request closed without merging**